# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: 7b3ae7465c92
# Full Hash: 7b3ae7465c9281d5674172961acd71a53507e7df
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-13 21:18:59
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add QuotaManager::PrincipalMetadataToPrincipalInfo method; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195570
# ==============================================================================

diff -r 092d5b1c73ba -r 7b3ae7465c92 dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Sun Oct 13 08:24:48 2024 +0000
+++ b/dom/quota/QuotaManager.h	Sun Oct 13 10:10:23 2024 +0000
@@ -589,6 +589,9 @@
   Result<PrincipalMetadata, nsresult> GetInfoFromValidatedPrincipalInfo(
       const PrincipalInfo& aPrincipalInfo);
 
+  static Result<PrincipalInfo, nsresult> PrincipalMetadataToPrincipalInfo(
+      const PrincipalMetadata& aPrincipalMetadata);
+
   static nsAutoCString GetOriginFromValidatedPrincipalInfo(
       const PrincipalInfo& aPrincipalInfo);
 
